Libs/GObject-2.0/Native/Classes/TypeDictionary.cs

```diff
@@ -75,9 +75,12 @@ internal static System.Type GetSystemType(Type gtype)
             // tree until we find a type that is. As all objects are
             // descended from GObject, we will eventually find a parent
             // type that is registered.
-            
+
             while (!_reverseTypeDict.TryGetValue(gtype, out sysType))
+            {
                 gtype = new Type(Functions.TypeParent(gtype.Value));
+                if (gtype.Value == 0) throw new Exception("Could not retrieve parent type - is the typeid valid?");
```

Comment:
We could add a method to the `Type` struct so this reads:

```cs
if (gtype.IsInvalid()) throw new Exception("Could not retrieve parent type - is the typeid valid?");
```

I think this makes the intend clearer.

Reply:
Looking at it again, if TypeParent returns zero, it simply means the call failed. So instead of wrapping it in a `Type`, maybe it should read something like this:

```cs
var result = Functions.TypeParent(gtype.Value);
if (result == 0) throw new GTypeLookupException();
gtype = new Type(result);
```

Reply:
Yes I like your 2nd thought a litte better than the first. In general I think both approaches are valid.

Nevertheless I don't want any "magic values" inside the code, even if it is the number zero as it does not express the intend and makes it hard to look for all places where we are interested in certain basic types.

In addition to my first suggestion I have some more:

 ```cs
var result = Functions.TypeParent(gtype.Value);
if (result == (nuint) BasicTypes.Invalid) throw new GTypeLookupException();
gtype = new Type(result);
```

 ```cs
//AsNuint is an extension method on the enum.
var result = Functions.TypeParent(gtype.Value);
if (result == BasicTypes.Invalid.AsNuint()) throw new GTypeLookupException(); 
gtype = new Type(result);
```

What do you think? Do you have another suggestion?

Reply:
We've decided on moving `BasicTypes` to Native and then using them in `Type`'s static values